cpp/ql/src/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -321,6 +354,13 @@ predicate localFlowStep(Node nodeFrom, Node nodeTo) { simpleLocalFlowStep(nodeFr
  */
 predicate simpleLocalFlowStep(Node nodeFrom, Node nodeTo) {
   simpleInstructionLocalFlowStep(nodeFrom.asInstruction(), nodeTo.asInstruction())
+  or
+  exists(LoadInstruction load |
+    // TODO: These can probably be getSourceValue() after #3112 is merged
```

Comment:
It's now merged (that's the line I meant to comment on a minute ago).

Reply:
Sadly my comment turned out to be incorrect. The flow in the following program is not captured by only following exact `Chi` operands:

```c
void sink(int *o);

struct B
{
  int *c;
  void set(int *c) { this->c = c; }
};

void f7(B *b)
{
  b->set(new int);
}

void f8()
{
  B *b = new B();
  f7(b);
  sink(b->c); // flow
}
```

since the load on `b->c` only is a total overlap.

Reply:
Then you'll want to merge #3097 and use `isResultConflated` here.